.github/copilot-instructions.md

```diff
@@ -0,0 +1,283 @@
+# GitHub Copilot Instructions for TypeORM
+
+This document provides guidance for GitHub Copilot when working with the TypeORM codebase.
+
+## Project Overview
+
+TypeORM is a TypeScript-based Object-Relational Mapping (ORM) library that supports multiple databases including MySQL/MariaDB, PostgreSQL, MS SQL Server, Oracle, SAP HANA, SQLite, MongoDB, and Google Spanner. It implements both Active Record and Data Mapper patterns and runs on Node.js, Browser, React Native, and Electron platforms.
+
+## Architecture & Structure
+
+### Core Components
+
+- **`src/data-source/`** - DataSource (formerly Connection) management
+- **`src/entity-manager/`** - Entity management and operations
+- **`src/repository/`** - Repository pattern implementation
+- **`src/query-builder/`** - SQL query building
+- **`src/decorator/`** - TypeScript decorators for entities, columns, relations
+- **`src/driver/`** - Database-specific drivers
+- **`src/metadata/`** - Entity metadata management
+- **`src/schema-builder/`** - Schema creation and migration
+- **`src/migration/`** - Database migration system
+- **`src/subscriber/`** - Event subscriber system
+- **`src/persistence/`** - Entity persistence logic
+
+### Design Patterns
+
+- **Active Record Pattern**: Entities have methods to save, remove, and query themselves
+- **Data Mapper Pattern**: Repositories handle entity persistence separately from business logic
+- **Decorator Pattern**: Extensive use of TypeScript decorators for metadata definition
+- **Builder Pattern**: QueryBuilder for constructing complex queries
+
+## Coding Standards
+
+### TypeScript Configuration
+
+- Target: ES2021+ with CommonJS modules
+- Strict mode enabled with some exceptions
+- Decorators: `experimentalDecorators` and `emitDecoratorMetadata` enabled
+- Required imports: `reflect-metadata` must be imported before TypeORM
```

Comment:
Not sure if this line is relevant 🤷🏻‍♂️
Also, not sure if the strict mode exceptions are relevant. Ideally, we would like to remove the typecheck/linting exceptions if possible. Some we added because the code had too many warnings.